addons/project/controllers/portal.py

```diff
@@ -142,22 +148,30 @@ def portal_my_tasks(self, page=1, date_begin=None, date_end=None, sortby=None, f
             step=self._items_per_page
         )
         # content according to pager and archive selected
+        if groupby == 'project':
+            order = "project_id, %s" % order  # force sort on project first to group by project in view
         tasks = request.env['project.task'].search(domain, order=order, limit=self._items_per_page, offset=pager['offset'])
         request.session['my_tasks_history'] = tasks.ids[:100]
+        if groupby != 'none':
+            groupedTasks = [request.env['project.task'].concat(*g) for k, g in groupbyelem(tasks, itemgetter('project_id'))]
+        else:
+            groupedTasks = [tasks]
 
         values.update({
             'date': date_begin,
             'date_end': date_end,
             'projects': projects,
-            'tasks': tasks,
+            'groupedTasks': groupedTasks,
```

Comment:
Well if people update the module without updating their template, everything crashes. You should be backward compatible, taking those options in mind:
 - people pull the code, relaunch server
 - people pull the code, do not relaunch server, refresh browsers (for JS modifications in stable)
 - people pull the code, relaunch server with an -u

I think there are issues here as templates and controllers are not fully backward compatible.

Templates should still accept a list of tasks, and handle a list of grouped elements in additions. Maybe we could separate in values tasks (old task list) and grouped_tasks that take precedence.